src/DocumentFormat.OpenXml.Framework/Equality/OpenXmlElementEqualityComparer.cs

```diff
@@ -0,0 +1,255 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using DocumentFormat.OpenXml.Framework;
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+
+namespace DocumentFormat.OpenXml
+{
+    internal sealed class OpenXmlElementEqualityComparer : IEqualityComparer<OpenXmlElement>
+    {
+        /// <summary>
+        /// Gets the options regulating how equality is defined.
+        /// </summary>
+        internal OpenXmlElementEqualityOptions Options { get; }
+
+        internal OpenXmlElementEqualityComparer(OpenXmlElementEqualityOptions options)
+        {
+            this.Options = options;
+        }
+
+        /// <summary>
+        /// Determines equality for two given <see cref="OpenXmlElement"/>.
+        /// </summary>
+        /// <param name="x">First object.</param>
+        /// <param name="y">Second object.</param>
+        /// <returns></returns>
+        public bool Equals(OpenXmlElement? x, OpenXmlElement? y)
+        {
+            if (ReferenceEquals(x, y))
+            {
+                return true;
+            }
+
+            if (x == null || y == null)
+            {
+                return false;
+            }
+
+            if (x.GetType() != y.GetType())
+            {
+                return false;
+            }
+
+            if (!this.Options.RequireParsed)
```

Comment:
This method shiuld delegate out to these additional checks if needed to keep it more readable

Reply:
I understood this comment as move all the options that change equation behaviour into their own method, hope that was what you meant.